### ISSUE TITLE
Ability to configure how are arguments are passed to external compiler

### ISSUE THREAD

**Issue by eriksvedang** [carp-lang/Carp#1226]
Different compilers require different flags, order of arguments, etc.
It would be super handy to be able to define this per compiler/platform, something like:

```clojure
(defbackend osx-tcc
  {"compiler" "tcc"
   "lib" "-l%"
   "include" "-i%"})
```

(very rough sketch)

**Comment by guberathome** [comment]
There already seems to be a project setting:
```
(Project.config "libflag" "...")
```

If these were to be added _after_ the file name in the compiler command, the major obstacle for using gcc should be gone.
Not having to configure profile.carp for this would indeed be _nice_to_have.

**Comment by eriksvedang** [comment]
Yeah, basically this would move all of those settings into a data structure and make it easy to switch between different sets of them.

**Comment by guberathome** [comment]
Just to illustrate my current issue with gcc (on Linux), which might provide additional motivation for the suggested solution:

Supposing you had any Carp source file like this...
```
; content of hello.carp 
(Project.config "echo-compiler-cmd" true)
(Project.config "compiler" "gcc")

(defn main []
	(IO.println &(fmt "Hello there: it took us only %d iterations to approximate e=%f" 123 2.71828 )) )
```

... then  the current output from Carp fails since ld can not link functions from libmath (or libm):
```
gcc  -o out/Untitled -I /home/ptokremin/.local/carp//core/  -fPIC -g -std=c99 -D_DEFAULT_SOURCE -Wall -Werror -Wno-unused-variable -Wno-self-assign -lm out/main.c
/usr/bin/ld: /tmp/ccisakli.o: in function `Double_acos':
/home/ptokremin/.local/carp//core/carp_double.h:68: undefined reference to `acos'
/usr/bin/ld: /tmp/ccisakli.o: in function `Double_asin':
/home/ptokremin/.local/carp//core/carp_double.h:72: undefined reference to `asin'
[...]
/usr/bin/ld: /tmp/ccisakli.o: in function `Float_ceil':
/home/ptokremin/.local/carp//core/carp_float.h:123: undefined reference to `ceilf'
/usr/bin/ld: /tmp/ccisakli.o: in function `Float_floor':
/home/ptokremin/.local/carp//core/carp_float.h:127: undefined reference to `floorf'
/usr/bin/ld: /tmp/ccisakli.o: in function `Float_mod':
/home/ptokremin/.local/carp//core/carp_float.h:131: undefined reference to `fmodf'
collect2: error: ld returned 1 exit status
carp: callCommand: gcc  -o out/Untitled -I /home/ptokremin/.local/carp//core/  -fPIC -g -std=c99 -D_DEFAULT_SOURCE -Wall -Werror -Wno-unused-variable -Wno-self-assign -lm out/main.c (exit 1): faileds
```

Fixing this problem is simply a matter of positioning the missing linker flag "-lm" behind the source file (in this case "out/main.c"):
```
gcc  -o out/Untitled -I /home/ptokremin/.local/carp//core/  -fPIC -g -std=c99 -D_DEFAULT_SOURCE -Wall -Werror -Wno-unused-variable -Wno-self-assign out/main.c -lm
```

So in a nutshell:
```
gcc  -o out/Untitled ... -lm out/main.c		
=> fails to compile

gcc  -o out/Untitled ... out/main.c  -lm
=> works
```

**Comment by eriksvedang** [comment]
@guberathome OK thanks, that's great context! We should make sure that the system is flexible enough to work around this in a really easy and clean manner.
